lib/sync-settings.coffee

```diff
@@ -68,29 +84,27 @@ module.exports =
         atom.notifications.addError "sync-settings: Error retrieving your settings. ("+message+")"
         return
 
-      settings = JSON.parse(res.files["settings.json"].content)
-      console.debug "settings: ", settings
-      @applySettings "", settings
+      for own file, filename of res.files
+        switch filename
+          when 'settings.json'
+            @applySettings '', JSON.parse(file.content)
+
+          when 'packages.json'
+            @installMissintPackages JSON.parse(file.content), cb
 
-      packages = JSON.parse(res.files["packages.json"].content)
-      console.debug "packages: ", packages
-      @installMissingPackages packages, cb
+          when 'keymap.cson'
+            fs.writeFileSync atom.keymap.getUserKeymapPath(), file.content
 
-      keymap = res.files['keymap.cson']?.content
-      console.debug "keymap.cson = ", res.files['keymap.cson']?.content
-      fs.writeFileSync(atom.keymap.getUserKeymapPath(), res.files['keymap.cson'].content) if keymap
+          when 'styles.less'
+            fs.writeFileSync atom.themes.getUserStyleSheetPath(), file.content
 
-      styles = res.files['styles.less']?.content
-      console.debug "styles.less = ", res.files['styles.less']?.content
-      fs.writeFileSync(atom.themes.getUserStylesheetPath(), res.files['styles.less'].content) if styles
+          when 'init.coffee'
+            fs.writeFileSync atom.config.configDirPath + "/init.coffee", file.content
 
-      initCoffee = res.files['init.coffee']?.content
-      console.debug "init.coffee = ", initCoffee
-      fs.writeFileSync(atom.config.configDirPath + "/init.coffee", initCoffee) if initCoffee
+          when 'snippets.cson'
+            fs.writeFileSync atom.config.configDirPath + "/snippets.coffee", file.content
 
-      snippetsCson = res.files['snippets.cson']?.content
-      console.debug "snippets.cson = ", snippetsCson
-      fs.writeFileSync(atom.config.configDirPath + "/snippets.cson", snippetsCson) if snippetsCson
+          else fs.writeFileSync "#{atom.config.configDirPath}/#{filename}"
```

Comment:
I didn't perform that check because the `extraFiles` config would be empty on a new install of Atom